server/src/main.rs

```diff
@@ -45,6 +46,27 @@ pub struct Context {
 
 /** Public Functions **/
 fn main() {
+    // A layer that logs events to a file.
+    let file = match File::create("debug.log") {
```

Comment:
I agree, there is no reason to add extra overhead to tracing with a filewrite.

Reply:
Fixed

Reply:
I think we should remove `.pretty()` and `.with_filter(filter::LevelFilter::INFO)` so the format of `stdout_log` matches the single line format the removed `debug_log` wrote to the file as mentioned above. By removing `.with_filter(filter::LevelFilter::INFO)` we can also statically control which log messages are printed completely from the features of the `tracing` crate in `cargo.toml` instead of both in `cargo.toml` and `main.rs`. If we want to have more control over which messages are printed in the future we can always add [`EnvFilter` ](https://docs.rs/tracing-subscriber/latest/tracing_subscriber/filter/struct.EnvFilter.html) so the filtering level can be set as an environment variable.

Reply:
Done